sdk/Sdk.Analyzers/BindingTypeNotSupported.cs

```diff
@@ -0,0 +1,133 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Diagnostics;
+
+namespace Microsoft.Azure.Functions.Worker.Sdk.Analyzers
+{
+    [DiagnosticAnalyzer(LanguageNames.CSharp)]
+    public class BindingTypeNotSupported : DiagnosticAnalyzer
+    {
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics { get { return ImmutableArray.Create(DiagnosticDescriptors.BindingTypeNotSupported); } }
```

Comment:
```suggestion
        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics { get; } = ImmutableArray.Create(DiagnosticDescriptors.BindingTypeNotSupported);
```

What you have now will create a new `ImmutableArray` on every `get` on this property - which is potentially a lot of extra objects and GC. The above syntax I suggested will instantiate the array once and return it for every `get`.

Reply:
Good catch. 

I think our existing analyzers are already doing that.

Reply:
Yeah I copied this from the other analyzers - will make a PR to update them all to fix this